src/vmod_soap.c

```diff
@@ -114,11 +118,11 @@ static struct priv_soap_task* init_task(VRT_CTX)
 /* -----------------------------------------------------------------/
    destroy session
 */
-static void clean_task(void *priv)
+static void clean_task(VRT_CTX, void *priv)
 {
 	struct priv_soap_task *soap_task;
 
-	AN(priv);
```

Comment:
shall we preserve the `AN(priv)` ?

Reply:
`CAST_OBJ_NOTNULL()` two lines down already asserts on the value not being NULL